### PR TITLE
Simplify compiling with pthreads when OpenMP is off

### PR DESCRIPTION
**Pull request by tkoskela**
Slightly less obscure way of ensuring we always pass `-pthread` when OpenMP is off. Inspired by https://github.com/astro-informatics/sopt/pull/264

### CODE REVIEW

**Review by SJaffa — COMMENTED** [review]
I find the Cmake files quite hard to follow. Should we add some more comments to try to explain what is going on?

**Comment by tkoskela** [comment]
You are not alone in that. I'll and comments on the parts I've changed/understood 😅

**Comment by tkoskela** [comment]
Does it make any more sense now?

**Review by SJaffa — APPROVED** [review]
